src/App.vue

```diff
@@ -1,54 +1,117 @@
-<template>
-  <div class="w-full h-full p-4 flex flex-col gap-4">
-    <h1 class="font-bold text text-gray-600 uppercase">
-      Zazuko Cube Viewer
-    </h1>
-    <source-config v-model:source="source" v-model:cubeUri="cubeUri" v-model:language="language" />
-    <cube-viewer
-      v-if="source && cubeUri"
-      :source="source"
-      :cubeUri="cubeUri"
-      :language="language"
-      class="bg-white rounded shadow-lg overflow-x-auto"
-    />
-  </div>
-</template>
-
-<script>
-import { defineComponent, ref } from 'vue'
-import CubeViewer from './components/CubeViewer.vue'
-import SourceConfig from './components/SourceConfig.vue'
+<script setup>
+/* eslint-disable */
+import { useUrlSearchParams } from '@vueuse/core'
+import { Parser } from 'n3'
 import { Source } from 'rdf-cube-view-query'
+import rdf from 'rdf-ext'
+import { onMounted, ref } from 'vue'
+import SourceConfig from './components/SourceConfig.vue'
+import Viewer from './components/Viewer.vue'
 
-const defaultLanguage = ['en', '*']
+const parser = new Parser()
+const params = useUrlSearchParams('history')
 
-export default defineComponent({
-  name: 'App',
-  components: { CubeViewer, SourceConfig },
+const source = ref(null)
+const viewInput = ref()
+const initError = ref()
 
-  setup () {
-    const urlParams = new URLSearchParams(window.location.search.substring(1))
+onMounted(() => {
 
-    const source = ref(null)
-    const endpointUrl = urlParams.get('endpointUrl')
-    if (endpointUrl) {
-      source.value = new Source({
-        endpointUrl,
-        user: urlParams.get('user') || undefined,
-        password: urlParams.get('password') || undefined,
-        sourceGraph: urlParams.get('sourceGraph') || undefined,
-      })
+  // Tries to parse a view
+  let dataset = undefined
+  if (params.view) {
+    try {
+      const quads = parser.parse(params.view)
+      dataset = rdf.dataset().addAll(quads)
+    } catch (e) {
+      initError.value = e
     }
+  }
 
-    const cubeUri = ref(urlParams.get('cube'))
+  if (params.endpointUrl) {
+    source.value = new Source({
+      endpointUrl: params.endpointUrl,
+      user: params.user,
+      password: params.password,
+      sourceGraph: params.sourceGraph
+    })
+  }
 
-    const language = ref(defaultLanguage)
-
-    return {
-      source,
-      cubeUri,
-      language,
-    }
-  },
+  viewInput.value = onlyOne({
```

Comment:
A view can be loaded in different ways: 
1. a dataset
2. a cube URI
3. a view URI